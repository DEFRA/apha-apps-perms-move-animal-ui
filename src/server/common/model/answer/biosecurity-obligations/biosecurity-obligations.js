import { HiddenAnswer } from '../hidden/hidden.js'

/** @import {HiddenAnswerConfig} from '../hidden/hidden.js' */

/**
 * export @typedef {'yes'} BiosecurityObligationsData
 * @typedef {{ biosecurityObligationsAcknowledged: BiosecurityObligationsData }} BiosecurityObligationsPayload
 */

/**
 * @augments {HiddenAnswer<BiosecurityObligationsPayload>}
 */
export class BiosecurityObligationsAnswer extends HiddenAnswer {
  /** @type {HiddenAnswerConfig} */
  static config = {
    payloadKey: 'biosecurityObligationsAcknowledged',
    value: 'yes'
  }
}
