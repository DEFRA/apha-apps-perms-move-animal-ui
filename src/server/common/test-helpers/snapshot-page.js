import { parseDocument } from '../../common/test-helpers/dom.js'
import { createServer } from '~/src/server/index.js'
import { withCsrfProtection } from './csrf.js'
/** @import { Server } from '@hapi/hapi' */

/* global expect, it, beforeAll, afterAll, describe */

/**
 * @param {string} describeMessage
 * @param {string} shouldMessage
 * @param {string} urlPath
 */
export const describePageSnapshot = (
  describeMessage,
  shouldMessage,
  urlPath
) => {
  describe(describeMessage, () => {
    /** @type {Server} */
    let server

    beforeAll(async () => {
      server = await createServer()
      await server.initialize()
    })

    afterAll(async () => {
      await server.stop({ timeout: 0 })
    })

    it(shouldMessage, async () => {
      const { payload } = await server.inject(
        withCsrfProtection({
          method: 'GET',
          url: urlPath
        })
      )

      const content =
        parseDocument(payload).querySelector('#main-content')?.innerHTML

      expect(content).toMatchSnapshot()
    })
  })
}
