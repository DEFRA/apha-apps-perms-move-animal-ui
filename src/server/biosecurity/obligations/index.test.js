import { BiosecurityObligationsAnswer } from '../../common/model/answer/biosecurity-obligations/biosecurity-obligations.js'
import { describePageSnapshot } from '../../common/test-helpers/snapshot-page.js'
import { keptSeparatelyPage } from '../kept-separately/index.js'
import { ObligationsPage, obligationsPage } from './index.js'

const sectionKey = 'biosecurity'
const questionKey = 'biosecurityObligationsAcknowledged'
const view = 'biosecurity/obligations/index.njk'
const pageUrl = '/biosecurity/biosecurity-intro'

describe('ObligationsPage', () => {
  const page = new ObligationsPage()

  it('should have the correct urlPath', () => {
    expect(page.urlPath).toBe(pageUrl)
  })

  it('should have the correct sectionKey', () => {
    expect(page.sectionKey).toBe(sectionKey)
  })

  it('should have the correct questionKey', () => {
    expect(page.questionKey).toBe(questionKey)
  })

  it('should have the correct view', () => {
    expect(page.view).toBe(view)
  })

  it('should have the correct Answer model', () => {
    expect(page.Answer).toBe(BiosecurityObligationsAnswer)
  })

  it('nextPage should return kept separately page', () => {
    const nextPage = page.nextPage()
    expect(nextPage).toBe(keptSeparatelyPage)
  })

  it('should export page', () => {
    expect(obligationsPage).toBeInstanceOf(ObligationsPage)
  })

  describePageSnapshot({
    describes: 'ObligationsPage.content',
    it: 'should render expected response and content',
    pageUrl
  })
})
