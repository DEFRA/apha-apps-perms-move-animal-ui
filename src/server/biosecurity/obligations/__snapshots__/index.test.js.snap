// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObligationsPage ObligationsPage.content should render expected response and content 1`] = `
"
  <h1 class="govuk-heading-l">Biosecurity on your farm or premises</h1>
  <div class="govuk-grid-row">
    <div class="govuk-grid-column-two-thirds">
      <p>
        You must answer the questions in this section as accurately as you can.
      </p>
      <p>
        It is important you take the time to consider your biosecurity measures
        and the steps you are taking to prevent the spread of TB.
      </p>
      <p>
        A vet will review your answers and decide on the outcome of your
        application.
      </p>
      <details class="govuk-details" data-module="govuk-details">
        <summary class="govuk-details__summary">
          <span class="govuk-details__summary-text">
            Guidance on biosecurity
          </span>
        </summary>
        <div class="govuk-details__text">
          <p>
            Read the GOV.UK guidance on
            <a target="_blank" href="https://www.gov.uk/guidance/protect-cattle-against-tb-infection-in-high-and-low-incidence-areas">
              how to protect cattle against TB infection</a>
            (opens in new tab).
          </p>
        </div>
      </details>
    </div>
  </div>
    <form method="POST" novalidate="">
      <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
          <input type="hidden" name="crumb" value="csrf-value">
          <input type="hidden" name="nextPage" value="">
<input type="hidden" name="biosecurityObligationsAcknowledged" value="yes">
            <button type="submit" class="govuk-button" data-module="govuk-button" id="continue-button">
  Continue
</button>
        </div>
      </div>
    </form>
        "
`;
